src/test/java/com/google/devtools/build/lib/remote/GrpcCacheClientTest.java

```diff
@@ -151,6 +154,51 @@ public void onError(Throwable t) {
     client.ensureInputsPresent(context, merkleTree, ImmutableMap.of(), /*force=*/ true);
   }
 
+  @Test
+  public void testChunkerResetAfterError() throws Exception {
+    GrpcCacheClient client = newClient();
+    serviceRegistry.addService(
+        new ByteStreamImplBase() {
+          @Override
+          public StreamObserver<WriteRequest> write(
+              StreamObserver<WriteResponse> responseObserver) {
+            return new StreamObserver<WriteRequest>() {
+              @Override
+              public void onNext(WriteRequest request) {
+                responseObserver.onError(Status.DATA_LOSS.asRuntimeException());
+              }
+
+              @Override
+              public void onCompleted() {}
+
+              @Override
+              public void onError(Throwable t) {}
+            };
+          }
+        });
+    byte[] data = new byte[20];
+    Digest digest = DIGEST_UTIL.compute(data);
+    CountDownLatch latch = new CountDownLatch(1);
```

Comment:
@coeuvre I see that when this PR landed in 5df79b6513c6dbb0fadf2e65fa95f37e2b7ab540, this part of the test was changed to have an `AtomicBoolean` `closed` flag instead of the latch I propose in this PR. The reason for the latch is that the `.close()` is called in a listener to the future returned by `uploadChunker`. Therefore, in theory, `.close()` could be called slightly after `.get()` on the future returns.

Reply:
I see. Thanks for the explanation. Will change it back.